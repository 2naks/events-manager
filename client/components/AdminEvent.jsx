import React, { Component } from 'react';
import moment from 'moment';

/**
 * React component for admin event
 * card displayed in the admin panel
 */
class AdminEvent extends Component {
  /**
   * React's method to render react component
   * @return {object}
   */
  render() {
    return (
      <div className="card-block">
        <div className="container events">
          <div className="row event-row">
            <div className="col-md-4 event">
              <div className="card">
                <div className="card-block">
                  <h5 className="card-title"><b>{this.props.event.title}</b></h5>
                  <p className="card-text h6">{this.props.event.description}</p>
                  <p className="card-text h6"><b>Venue</b>: {this.props.event.location}</p>
                  <p className="card-text h6"><b>Date</b>: {moment(this.props.event.date).format('DD MMMM YYYY')}</p>
                  <p className="card-text h6"><b>Time</b>: { this.props.event.time }</p>
                  <p className="card-text h6"><b>Status</b>: Holding</p>
                  <p className="text-center">
                    <a href={`/events/${this.props.event.id}`} className="btn btn-sm btn-primary">
                      <i className="fa fa-info-circle fa-lg" /> View Event
                    </a>
                    <a href="#" className="btn btn-sm btn-danger" ><i className="fa fa-times fa-lg" /> Cancel</a></p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default AdminEvent;
