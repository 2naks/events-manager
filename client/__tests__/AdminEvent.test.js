import AdminEvent from '../components/AdminEvent.jsx';

describe('AdminEvent Component', () => {
  const props = {
    event: {
      id: "1",
      title: "",
      description: "",
      location: "",
      date: "",
      time: "",
      status: 'true'
    },
    show: () => {}
  };

  it('it should render component shallow', () => {
    const wrapper = shallow(<AdminEvent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render AdminEvent component mounted', () => {
    const wrapper = mount(<AdminEvent {...props} />);
    expect(wrapper.length).toEqual(1);
  });

  it('it should handle onClick event', () => {
    const wrapper = shallow(<AdminEvent {...props} />);
    let viewEventButton = wrapper.find('.btn-primary');
    viewEventButton.simulate('click');
  });
});
