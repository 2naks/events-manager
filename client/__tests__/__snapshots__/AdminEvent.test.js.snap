// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminEvent Component it should render component shallow 1`] = `
<div
  className="card-block"
>
  <div
    className="container events"
  >
    <div
      className="row event-row"
    >
      <div
        className="col-md-4 event"
      >
        <div
          className="card"
        >
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b />
            </h5>
            <p
              className="card-text h6"
            />
            <p
              className="card-text h6"
            >
              <b>
                Venue
              </b>
              : 
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Date
              </b>
              : 
              Invalid date
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Time
              </b>
              : 
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Status
              </b>
              : Holding
            </p>
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/events/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Event
              </a>
              <a
                className="btn btn-sm btn-danger"
                href="#"
              >
                <i
                  className="fa fa-times fa-lg"
                />
                 Cancel
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
